server/moodle/mod/livequiz/quizcreator/save_quiz.php

```diff
@@ -0,0 +1,81 @@
+<?php
+// This file is part of Moodle - https://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <https://www.gnu.org/licenses/>.
+
+/**
+ * Retrieves required parameters, validates user authentication,
+ * and saves quiz questions to the database.
+ * @package   mod_livequiz
+ * @copyright 2024 Software AAU
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+require_once('../../../config.php');
+
+global $DB;
+
+
+
+// Get course ID from parameters.
+$id = required_param('id', PARAM_INT);
+
+require_login();
+
+if ($_SERVER['REQUEST_METHOD'] === 'POST') {
+    $inputdata = file_get_contents("php://input");
+    $data = json_decode($inputdata);
+
+    if ($data && !empty($data->questions)) {
+        $quizdata = new stdClass();
+        $quizdata->id = $data->id;
+        $quizdata->name = $data->name;
+        $quizdata->intro = $data->intro;
+        $quizdata->introformat = FORMAT_HTML; // Assuming its HTML, change if I am wrong.
+        $quizdata->timemodified = $data->timemodified;
+        $quizdata->timecreated = $data->timecreated;
+
+        try {
+            $quizid = $DB->insert_record('livequiz', $quizdata);
```

Comment:
This should not be handled by the quizcreator. This should __only__ be handled by the model layer, through interaction with the service layer

See this for context:
https://github.com/AAU-P5-Moodle/moodle-1/blob/staging/server/moodle/mod/livequiz/classes/services/livequiz_services.php

the ```submit_quiz``` handles this logic for you

Please reach out if in doubt <3

Reply:
okay, i was unaware, will get this fixed

Reply:
Lovely
Igen, giv lyd. Skal gladeligt give et rundown af det